src/controller/CHIPDeviceController.h

```diff
@@ -141,6 +141,20 @@ struct ControllerInitParams
      */
     bool removeFromFabricTableOnShutdown = true;
 
+    /**
+     * Controls whether use the storage fabric table. If operational key pair
+     * or NOC chain are not passed, the fabric index are not available even if
+     * there is a fabric in storage. This provides a way to SetupController when
+     * the device has NOC installed and reboots, by providing a storage fabric
+     * index instead of operational key pair and NOC chain.
+     */
+    bool enableUseStorageFabric = false;
+
+    /**
+     * Use a storage fabric index to find the fabric table.
+     */
+    FabricIndex useStorageFabricIndex = kUndefinedFabricIndex;
```

Comment:
`use` prefix sounds like a bool. maybe just `storageFabricIndex` or something similar. I am still somewhat unclear about the usage here.

Reply:
Use `chip::Optional<FabricIndex> storageFabricIndex` instead.
It is mainly used to SetupController without operational key pair and NOC chain  params for a device which have installed NOC before.